code/game/objects/items/devices/transfer_valve.dm

```diff
@@ -128,22 +128,24 @@
 			toggle = 1
 
 /obj/item/transfer_valve/on_update_icon()
-	overlays.Cut()
+	. = ..()
 	underlays.Cut()
 
 	if(!tank_one && !tank_two && !attached_device)
 		icon_state = "valve_1"
 		return
 	icon_state = "valve"
 
+	var/list/cur_overlays
 	if(tank_one)
-		overlays += "[tank_one.icon_state]"
+		LAZYADD(cur_overlays, new /mutable_appearance(tank_one))
 	if(tank_two)
 		var/icon/J = new(icon, icon_state = "[tank_two.icon_state]")
 		J.Shift(WEST, 13)
 		underlays += J
 	if(attached_device)
-		overlays += "device"
+		LAZYADD(cur_overlays, overlay_image(icon, "device"))
+	add_overlay(cur_overlays)
```

Comment:
As mentioned on Discord, SSoverlays implicitly avoid appearance churn, there's no benefit to collecting overlays before adding them.

Reply:
I removed needlessly creating lists for adding overlays in a bunch of places.